plugin/sampling/strategystore/adaptive/factory.go

```diff
@@ -83,12 +84,22 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, ssFactory storage.S
 
 // CreateStrategyStore implements strategystore.Factory
 func (f *Factory) CreateStrategyStore() (strategystore.StrategyStore, strategystore.Aggregator, error) {
-	p, err := NewStrategyStore(*f.options, f.metricsFactory, f.logger, f.lock, f.store)
+	participant := leaderelection.NewElectionParticipant(f.lock, defaultResourceName, leaderelection.ElectionParticipantOptions{
```

Comment:
you are passing the same elector to two components, and each of them seem to call Start on it. Usually we initialize resources in Factory.Initialize, not in Create methods.